File: code/demonstration/receiver/AndroidIteration1/app/src/main/java/de/fzi/sensidl/receiver/android/iteration1/MainActivity.java
package de.fzi.sensidl.receiver.android.iteration1;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import java.util.Random;

import de.fzi.sensidl.R;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

	/** String for a clear key to save the shown random number on orientation change */
	private static final String RANDOM_NUMBER_ARGUMENT = "randomNumber";

	private Button   calculateRandomNumber;
	private TextView randomNumber;

	/**
	 * The onCreate method called to create the view
	 * Find the views and set an on click listener on the button
	 * @param savedInstanceState
	 */
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.main_activity);

		calculateRandomNumber = (Button) findViewById(R.id.calculate_random_number);
		randomNumber = (TextView) findViewById(R.id.random_number);

		calculateRandomNumber.setOnClickListener(this);
	}

	/**
	 * Handle an onClick event
	 * @param v The view which was clicked
	 */
	@Override
	public void onClick(View v) {
		switch (v.getId()) {
			case R.id.calculate_random_number:
				calculateAndShowRandomNumber();
				break;
		}
	}

	/**
	 * Calculate a random number and show it.
	 */
	private void calculateAndShowRandomNumber() {
		Random random = new Random();
		randomNumber.setText(String.valueOf(random.nextInt()));
	}

	/**
	 * Save the shown random number.
	 * @param outState
	 */
	@Override
	protected void onSaveInstanceState(Bundle outState) {
		super.onSaveInstanceState(outState);
		outState.putString(RANDOM_NUMBER_ARGUMENT, randomNumber.getText().toString());
	}

	/**
	 * Restore the saved random number and show it.
	 * If there is no saved random number show default "".
	 * @param savedInstanceState
	 */
	@Override
	protected void onRestoreInstanceState(Bundle savedInstanceState) {
		super.onRestoreInstanceState(savedInstanceState);
		randomNumber.setText(savedInstanceState.getString(RANDOM_NUMBER_ARGUMENT, ""));
	}
}

File: code/demonstration/receiver/XamarinIteration1/XamarinStufe1/XamarinStufe1.Droid/MainActivity.cs
using Android.App;
using Android.OS;
using Android.Support.V7.App;
using Android.Views;
using Android.Widget;

namespace XamarinIteration1.Droid
{
	// <summary>
	// The main activity for the android project
	// </summary>
	[Activity (Label = "@string/app_name", MainLauncher = true, Icon = "@drawable/icon")]
	public class MainActivity : AppCompatActivity, View.IOnClickListener
	{
		private const string RandomNumberArgument = "randomNumber";

		private Button _calculateRandomNumber;
		private TextView _randomNumber;

		// <summary>
		// The onCreate method called to create the view
		// Find the views and set an on click listener on the button
		// </summary>
		// <param name="bundle">bundle</param>
		protected override void OnCreate (Bundle bundle)
		{
			base.OnCreate (bundle);
			SetContentView (Resource.Layout.Main);
			_calculateRandomNumber = FindViewById<Button> (Resource.Id.calculate_random_number);
			_randomNumber = FindViewById<TextView> (Resource.Id.random_number);

			_calculateRandomNumber.SetOnClickListener (this);
		}
			
		// <summary>
		// Handle an onClick event
		// </summary>
		// <param name="v">the view</param>
		public void OnClick (View v)
		{
			switch (v.Id) {
			case Resource.Id.calculate_random_number:
				CalculateAndShowRandomNumber ();
				break;
			}
		}
			
		// <summary>
		// Calculate a random number and show it.
		// </summary>
		private void CalculateAndShowRandomNumber ()
		{
			_randomNumber.Text = GenerateRandomNumberClass.GetRandomNumber ().ToString ();
		}
			
		// <summary>
		// Save the shown random number.
		// </summary>
		// <param name="outState">outState</param>
		protected override void OnSaveInstanceState (Bundle outState)
		{
			base.OnSaveInstanceState (outState);
			outState.PutString (RandomNumberArgument, _randomNumber.Text);
		}
			
		// <summary>
		// Restore the saved random number and show it.
		// If there is no saved random number show default "".
		// </summary>
		// <param name="savedInstanceState">savedInstanceState</param>
		protected override void OnRestoreInstanceState (Bundle savedInstanceState)
		{
			base.OnRestoreInstanceState (savedInstanceState);
			_randomNumber.Text = savedInstanceState.GetString (RandomNumberArgument, "");
		}
	}
}